osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/utils/FileZipper.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.shared.utils;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.zip.Deflater;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Compress a file to a zip file. Use the default constructor
+ * {@link FileZipper#FileZipper()} to create an instance with
+ * {@link Deflater.BEST_COMPRESSION} compression level and input file deletion
+ * after the file has been zipped.
+ *
+ * Alternatively, choose a compression level using constructor
+ * {@link FileZipper#FileZipper(int, boolean) by passing a constant defined by
+ * the class {@link Deflater}}. Disable input file deletion if desired.
+ */
+public class FileZipper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileZipper.class);
+
+    private static final int ONE_MB = 1 * 1024 * 1024;
+    private static final String ZIP = ".zip";
+
+    public static String compressFileUsingDefaultSettings(final String filePath) throws IOException {
+        final FileZipper fileZipper = new FileZipper();
+        return fileZipper.compressFile(filePath);
+    }
+
+    private final int compressionLevel;
+
+    private final boolean removeInputFile;
+
+    public FileZipper() {
+        this.compressionLevel = Deflater.BEST_COMPRESSION;
+        this.removeInputFile = true;
+    }
+
+    public FileZipper(final int compressionLevel, final boolean removeInputFile) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1213)